### PR TITLE
Tweak ci dependency cache

### PR DESCRIPTION
**Pull request by wemeetagain**

### CODE REVIEW

**Review by mpetrunic — DISMISSED** [review]

**Comment by dapplion** [comment]
Running spec tests with `mocha --parallel --jobs 2`
|   |  serial | parallel  |
|---|---|---|
| spec min step | 1m 8s | 1m 14s  |
| spec min job | 5m 11s | 5m 33s  |
| spec fast step  | 16m 10s  | 8m 55s |
| spec fast job | 19m 45s | 13m 44s |

-----

The slowest test by far is `finality fast`. Total 493409ms = 8m 13s
```
@chainsafe/spec-test-runner:   finality fast
@chainsafe/spec-test-runner:     ✓ finality_no_updates_at_genesis (69650ms)
@chainsafe/spec-test-runner:     ✓ finality_rule_1 (100798ms)
@chainsafe/spec-test-runner:     ✓ finality_rule_2 (100894ms)
@chainsafe/spec-test-runner:     ✓ finality_rule_3 (164318ms)
@chainsafe/spec-test-runner:     ✓ finality_rule_4 (57749ms)
```
And mocha parallelizes by file, so this file is the limiting factor. Can this file be split so mocha can parallelize these ones too?

**Comment by dapplion** [comment]
@wemeetagain About space requirements for the spec tests, here is `df` data captured after each step in the spec-fast job run 

**without running the free-space script**

| CI step        | Used | Available |
|----------------|----------|----------|
| checkout       | 65704228 | 21497512 |
| Setup-node     | 65899728 | 21302012 |
| Restor deps    | 65899752 | 21301988 |
| Install + Build  | 78936108 | 8265632  |
| test:spec-fast | 78936316 | 8265424  |

**running free space script**

| CI step        | Used | Available |
|----------------|----------|----------|
| checkout       | 65704228 | 21497512 |
| Setup-node     | 65899728 | 21302012 |
| Free space | 61929044 | 25272696 |
| Restor deps    | 68347192 | 18854548 |
| Build  | 68367104 | 18834636  |
| test:spec-fast | 68367884 | 18833856  |

- Install + build spec deps: +13GB
- Restore spec deps: +6.4GB
- free space: -4GB

With these numbers, free-ing up space does not seem necessary

File: .github/workflows/test-sim.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+          path: |
+            node_modules
+            packages/*/node_modules
+          key: ${{ runner.os }}-${{ hashFiles('**/yarn.lock') }}-ignore-optional
+      - name: Install & build
+        if: steps.cache-deps.outputs.cache-hit != 'true'
+        run: yarn install --frozen-lockfile --ignore-optional
+      - name: Build
+        run: node_modules/.bin/lerna run build
+        if: steps.cache-deps.outputs.cache-hit == 'true'
```
Comment (mpetrunic) [inline .github/workflows/test-sim.yml:28]: Aren't you suppose to run build always. Even though deps aren't changed, you could change code which isn't hashed by lock file afaik

File: .github/workflows/test-spec.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+          path: |
+            node_modules
+            packages/*/node_modules
+          key: ${{ runner.os }}-${{ hashFiles('**/yarn.lock') }}
+      - name: Install & build
+        if: steps.cache-deps.outputs.cache-hit != 'true'
+        run: yarn install --frozen-lockfile
+      - name: Build
+        run: node_modules/.bin/lerna run build
+        if: steps.cache-deps.outputs.cache-hit == 'true'
```
Comment (mpetrunic) [inline .github/workflows/test-spec.yml:30]: same here

File: .github/workflows/test-sim.yml (hunk 1 above)
Comment (dapplion) [reply .github/workflows/test-sim.yml:28]: `yarn install` runs build as a post install step

**Review by dapplion — APPROVED** [review]

**Pull request merged**